geopandas/geodataframe.py

```diff
@@ -338,12 +342,34 @@ def set_geometry(self, col, drop=False, inplace=False, crs=None):
                     "the column name is shared by multiple columns."
                 )
 
-            if drop and geo_column_name in frame.columns:
-                # if _geometry_column_name is None, geo_column_name
-                # may not exist
-                del frame[geo_column_name]
+            given_colname_drop_msg = (
+                "The `drop` keyword argument is deprecated and in future the only "
+                "supported behaviour will match drop=False. To silence this "
+                "warning and adopt the future behaviour, stop providing "
+                "`drop` as a keyword to `set_geometry`. To replicate the "
+                "`drop=True` behaviour you should update "
+                "your code to\n`geo_col_name = gdf.active_geometry_name;"
+                " gdf.set_geometry(new_geo_col).drop("
+                "columns=geo_col_name).rename_geometry(geo_col_name)`."
+            )
 
-            geo_column_name = col
+            if drop is False:  # specifically False, not falsy i.e. None
+                # User supplied False explicitly, but arg is deprecated
+                warnings.warn(
+                    given_colname_drop_msg,
+                    category=FutureWarning,
+                    stacklevel=2,
+                )
+            if drop:
+                del frame[col]
```

Comment:
Should this be `del frame[geo_column_name]`? (that was done in the previous iteration of this PR) 
And in that case should we keep the `and geo_column_name in frame.columns:` here before calling?